app/controllers/bulk_upload_sales_logs_controller.rb

```diff
@@ -24,12 +25,26 @@ def update
 
 private
 
-  def validate_data_protection_agrement_signed!
+  def validate_data_protection_agreement_signed!
     return if @current_user.organisation.data_protection_confirmed?
 
     redirect_to sales_logs_path
   end
 
+  def validate_year!
+    return if params[:id] == "year"
+    return if params[:id] == "guidance" && params.dig(:form, :year).nil?
+
+    allowed_years = [current_year]
+    allowed_years.push(current_year - 1) if FormHandler.instance.sales_in_crossover_period?
+    allowed_years.push(current_year + 1) if FeatureToggle.allow_future_form_use?
```

Comment:
We'd never have set up anything for years further ahead than this, so allowing just the one seems reasonable.

Reply:
Would it be useful extracting it into a FormHandler method at all? Something similar to `years_of_available_sales_forms`?